module_3/gift_certificates_system/service/src/main/java/com/epam/esm/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,141 @@
+        this.userDao = userDao;
+        this.gcDao = gcDao;
+    }
+
+    @Override
+    public OrderDto create(long userId, long giftCertificateId) {
+        // Find requested User in the database
+        User user = getUserOrElseThrow(userId);
+        UserDto userDto = modelMapper.map(user, UserDto.class);
+
+        // Find requested Gift certificate in the database
+        GiftCertificate certificate = getGiftCertificateOrElseThrow(giftCertificateId);
+        GiftCertificateDto certificateDto = modelMapper.map(certificate, GiftCertificateDto.class);
+
+        // Create OrderDto
+        OrderDto orderDto = new OrderDto();
+        orderDto.setUser(userDto);
+        orderDto.setGiftCertificate(certificateDto);
+
+        // Create Order entity with User and Gift certificate
+        Order order = modelMapper.map(orderDto, Order.class);
+        BigDecimal gcPrice = certificate.getPrice();
+        order.setPrice(gcPrice);
+        order.setUser(user);
+        order.setGiftCertificate(certificate);
+
+        return this.build(orderDao.create(order));
+    }
+
+    @Override
+    public Set<OrderDto> findAll(EsmPagination pagination) {
+        return orderDao.findAll(pagination, Order.class).stream()
+                .map(this::build)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public Set<OrderDto> findAllByUserId(long userId, EsmPagination pagination) {
+        User user = getUserOrElseThrow(userId);
+        return orderDao.findAllBy(user, pagination).stream()
+                .map(this::build)
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public OrderDto findOrderForUser(long orderId, long userId) {
+        User user = getUserOrElseThrow(userId);
+        Order order = getOrderOrElseThrow(orderId);
+
+        if (Objects.equals(user, order.getUser())) {
+            return build(order);
+        } else {
+            throw new EntityNonExistentException(EXCEPTION_ORDER_FOR_USER_NOT_FOUND, userId);
+        }
+    }
+
+    @Override
+    public OrderDto findById(long id) {
+        Order order = getOrderOrElseThrow(id);
+        return this.build(order);
+    }
+
+    @Override
+    public void delete(long id) {
+        throw new UnsupportedOperationException(EXCEPTION_UNSUPPORTED_OPERATION);
+    }
+
+    private User getUserOrElseThrow(long userId) {
+        return userDao.findById(userId).orElseThrow(() -> new EntityNonExistentException(EXCEPTION_USER_ID_NOT_FOUND, userId));
+    }
+
+    private Order getOrderOrElseThrow(long orderId) {
+        return orderDao.findById(orderId).orElseThrow(() -> new EntityNonExistentException(EXCEPTION_ORDER_ID_NOT_FOUND, orderId));
+    }
+
+    private GiftCertificate getGiftCertificateOrElseThrow(long gcId) {
+        return gcDao.findById(gcId).orElseThrow(() -> new EntityNonExistentException(EXCEPTION_GIFT_CERTIFICATE_ID_NOT_FOUND, gcId));
+    }
+
+    private OrderDto build(Order order) {
```

Comment:
please rename this method to something more specific, on example "mapEntityToDto" or something like this

Reply:
Thanks.
After refactoring `OrderServiceImpl.java` this method was removed.